transport/tlscommon/server_config_test.go

```diff
@@ -173,3 +173,80 @@ func Test_ServerConfig_Repack(t *testing.T) {
 		})
 	}
 }
+
+func Test_ServerConfig_RepackJSON(t *testing.T) {
```

Comment:
I'm not sure if we encounter json serialization in the error we observed but i've added tests in case that is the issue.
I've also added Unpack unit tests to all the types